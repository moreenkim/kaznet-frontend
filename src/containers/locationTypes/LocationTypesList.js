// Smart component that renders the LocationTypes List view
import React, { Component } from "react";
import { connect } from "react-redux";
import { Link } from "react-router-dom";
import { bindActionCreators } from "redux";

import * as locationTypeActions from "../../store/locationTypes/actions";
import * as locationTypeSelectors from "../../store/locationTypes/reducer";
import * as globalActions from "../../store/global/actions";

import ListView from "../../components/ListView";
import ElementMap from "../ElementMap";

export class LocationTypesList extends Component {
  componentDidMount() {
    this.props.showListTitle();
    this.props.fetchLocationTypes();
    this.props.changePageTitle("Location Types");
    this.props.changePageTitleButton("+ Create Location Type");
    this.props.changePageTarget("/locationTypes/new");
  }

  render() {
    if (!this.props.rowsById) return this.renderLoading();
    return (
      <div className="LocationTypeList">
        <ListView
          renderHeaders={this.renderHeaders}
          rowsIdArray={this.props.rowsIdArray}
          rowsById={this.props.rowsById}
          renderRow={this.renderRow}
        />
      </div>
    );
  }

  renderLoading() {
    return <p>Loading...</p>;
  }

  renderRow(row) {
    const rowItems = [
      <Link to={`/locationTypes/edit/${row.id}`} key="link_to">
        {row.attributes.name}
      </Link>
    ];
    return <ElementMap items={rowItems} HTMLTag="td" />;
  }

  renderHeaders() {
    const headerItems = ["Name"];
    return <ElementMap items={headerItems} HTMLTag="th" />;
  }
}

function mapStateToProps(state) {
  return {
    rowsById: locationTypeSelectors.getLocationTypesById(state),
    rowsIdArray: locationTypeSelectors.getLocationTypesIdArray(state)
  };
}

function mapDispatchToProps(dispatch) {
  return bindActionCreators(
    {
      fetchLocationTypes: locationTypeActions.fetchLocationTypes,
      changePageTitle: globalActions.changePageTitle,
      changePageTitleButton: globalActions.changePageTitleButton,
      showListTitle: globalActions.toggleDetailTitleOff,
      changePageTarget: globalActions.changePageTarget
    },
    dispatch
  );
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(LocationTypesList);
