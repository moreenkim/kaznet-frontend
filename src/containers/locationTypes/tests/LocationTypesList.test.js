import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Router } from 'react-router';
import createBrowserHistory from 'history/createBrowserHistory';
import { LocationTypesList } from '../LocationTypesList';
import * as fixtures from '../../../store/locationTypes/tests/fixtures';

const history = createBrowserHistory();

describe('containers/locationTypes/LocationTypesList', () => {
  it('renders without crashing', () => {
    shallow(
      <LocationTypesList
        fetchLocationTypes={function() {}}
        changePageTitle={function() {}}
        changePageTarget={function() {}}
        changePageTitleButton={function() {}}
        changePageNumber={function() {}}
        searchVal={function() {}}
        showListTitle={function() {}}
        location={history.location}
        rowsIdArray={fixtures.locationTypesIdArray}
      />
    );
  });

  it('renders location list correctly', () => {
    const wrapper = mount(
      <Router history={history}>
        <LocationTypesList
          fetchLocationTypes={function() {}}
          changePageTitle={function() {}}
          changePageTarget={function() {}}
          changePageTitleButton={function() {}}
          changePageNumber={function() {}}
          searchVal={function() {}}
          showListTitle={function() {}}
          rowsById={fixtures.locationTypesById}
          rowsIdArray={fixtures.locationTypesIdArray}
          endpoint="locationTypes"
          pageLinks={fixtures.pageLinks}
          totalPages={fixtures.totalPages}
          currentPage={fixtures.currentPage}
          firstPage={fixtures.firstPage}
          lastPage={fixtures.lastPage}
          location={history.location}
        />
      </Router>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.unmount();
  });
});
