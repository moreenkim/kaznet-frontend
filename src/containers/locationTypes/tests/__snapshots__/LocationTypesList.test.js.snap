// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/locationTypes/LocationTypesList renders location list correctly 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <div
    className="LocationTypeList"
  >
    <ListView
      renderHeaders={[Function]}
      renderRow={[Function]}
    >
      <div>
        <Table
          bordered={true}
          className="kaznet-table"
          responsiveTag="div"
          tag="table"
        >
          <table
            className="kaznet-table table table-bordered"
          >
            <thead>
              <tr>
                <ElementMap
                  HTMLTag="th"
                  items={
                    Array [
                      "Name",
                    ]
                  }
                >
                  <th
                    key="0"
                  >
                    Name
                  </th>
                </ElementMap>
              </tr>
            </thead>
            <tbody />
          </table>
        </Table>
      </div>
    </ListView>
  </div>
</LocationTypesList>
`;
